### PR TITLE
primary_site based subtyping

### PR DESCRIPTION
**Pull request by kgaonkar6**
#### What scientific question is your analysis addressing?
Subtype Neurocytoma tumors as central or extra-ventricular

#### What was your approach?
The subtyping is only based on primary_site from pbta-histologies.tsv and pathology_diagnosis == "Neurocytoma"

If primary_site == "Ventricles", then molecular_subtype == CNC (central neurocytoma) and if primary_site != "Ventricles", then molecular_subtype == EVN (extraventricular neurocytoma)

#### What GitHub issue does your pull request address?
#805 

#### Which areas should receive a particularly close look?
There was 1 sample with short_histology== `CNS Neuroblastoma` and pathology_diagnosis == `Neurocytoma` which I've included in the neurocytoma subtyping. Does that sound ok? 
I decided to keep it since the pathology_diagnosis and pathology_diagnosis_free_text `neurocytoma` and seems like it's not been updated from subtyping as well ( checked for Notes/molecular_subtype  which are NA )

#### Is there anything that you want to discuss further?
Looks like other subtyping results only have the Kids_First IDs and molecular_subtype , so I've kept just the basic columns. Should I add the integrated_diagnosis column as well or will that be part of the `compile_subtyping` step of molecular_subtype_pathology module?

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
yes

#### What types of results are included (e.g., table, figure)?
tables

#### What is your summary of the results?
results/neurocytoma_subtyping.tsv

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/molecular-subtyping-neurocytoma/01-neurocytoma-subtyping.Rmd [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+```{r}
+neurocytoma_from_pathology %>%
+  filter(short_histology == "CNS neuroblastoma"
+                    ) %>%
+  select(pathology_diagnosis,pathology_free_text_diagnosis)
+```
+
+
+Both pathology_diagnosis and pathology_free_text_diagnosis is Neurocytoma so keeping it in Neurocytoma list
```
Comment (jharenza) [inline analyses/molecular-subtyping-neurocytoma/01-neurocytoma-subtyping.Rmd:60]: Hi @kgaonkar6 thanks for this! Few comments and suggestions here:
1. I would suggest taking the search for `short_histology` out of this completely, as we are now relying solely on `pathology_diagnosis` and `pathology_free_text_diagnosis` for subtyping. Nice, check though :).
2. How many samples contain `Neurocytoma` in `pathology_diagnosis`? You are also assessing `pathology_free_text_diagnosis` for samples which have `Neurocytoma` in `pathology_diagnosis`. Are there any Neurocytoma samples in `pathology_free_text_diagnosis` that are not listed in `pathology_diagnosis`? (I checked and there is a 1:1 relationship here, but would be a good check to make sure we don't expect any in that `pathology_free_text_diagnosis` that are not in `pathology_diagnosis`.
3. Can you print the table of all primary site just to be sure there are no multiples containing `Ventricles;Other site` for example, in which case the != would not work. I know the answer, just making sure we are justifying steps as we go :).

File: analyses/molecular-subtyping-neurocytoma/01-neurocytoma-subtyping.Rmd [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+```{r}
+
+neurocytoma_subtype <- neurocytoma_from_pathology %>% 
+  dplyr::mutate(molecular_subtype= 
+                  case_when(primary_site == "Ventricles" ~ "CNC",
+                            primary_site != "Ventricles" ~ "EVN"
+                            )
```
Comment (jharenza) [inline analyses/molecular-subtyping-neurocytoma/01-neurocytoma-subtyping.Rmd:68]: Add code comment here about assigning subtypes at this step.

**Review by jharenza — APPROVED** [review]
Requested a few minor changes, but approve once those are in - results as expected, looks good!

File: analyses/molecular-subtyping-neurocytoma/01-neurocytoma-subtyping.Rmd (hunk 2 above)
Comment (kgaonkar6) [reply analyses/molecular-subtyping-neurocytoma/01-neurocytoma-subtyping.Rmd]: Thanks for the review @jharenza ,added comments here!

File: analyses/molecular-subtyping-neurocytoma/01-neurocytoma-subtyping.Rmd (hunk 1 above)
Comment (kgaonkar6) [reply analyses/molecular-subtyping-neurocytoma/01-neurocytoma-subtyping.Rmd]: I've added the checks for primary_site values and checking if there are any "Neurocytoma" samples in pathology_free_text_diagnosis that are not listed in pathology_diagnosis.
Thanks!

File: analyses/molecular-subtyping-neurocytoma/01-neurocytoma-subtyping.Rmd (hunk 2 above)
Comment (jharenza) [reply analyses/molecular-subtyping-neurocytoma/01-neurocytoma-subtyping.Rmd]: great, thank you!

File: analyses/molecular-subtyping-neurocytoma/01-neurocytoma-subtyping.Rmd (hunk 1 above)
Comment (jharenza) [reply analyses/molecular-subtyping-neurocytoma/01-neurocytoma-subtyping.Rmd]: looks good!

**Review by cbethell — COMMENTED** [review]
Took a look at this PR and it looks good!

I just left a few suggestions below pertaining to general structure of the notebook, but overall I was able to run the shell script to reproduce the results and it seems to satisfy #805 👍 

One other thing and that is, do we want the analysis notebook in this PR added to the `.circleci/config.yml` file? I believe the answer is yes, but I may have missed a discussion about this that would say otherwise.

File: analyses/molecular-subtyping-HGG/run-molecular-subtyping-HGG_003_008.sh [hunk 3]
```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-HGG/run-molecular-subtyping-HGG_003_008.sh:1]: My guess is that this is something you were working with locally having to do with PNOC 003 and 008 and that it should not be added here?

File: analyses/molecular-subtyping-HGG/input/cell-line-composition.tsv [hunk 4]
```diff
@@ -31,3 +31,7 @@ PT_Y5KY6KN9	BS_XMP9XNR9	A	7316-195	739834
 PT_Y5KY6KN9	BS_JGKRN7NA	S	7316-195	739835
 PT_Z4PJA6KT	BS_853PNV7P	A	7316-1763	739837
 PT_Z4PJA6KT	BS_TF5TTEXH	S	7316-1763	739836
+<<<<<<< HEAD
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-HGG/input/cell-line-composition.tsv:34]: These conflicts need to be resolved.

File: analyses/molecular-subtyping-HGG/run-molecular-subtyping-HGG_003_008.sh (hunk 3 above)
Comment (kgaonkar6) [reply analyses/molecular-subtyping-HGG/run-molecular-subtyping-HGG_003_008.sh:1]: yeahh I committed the wrong files here by mistake.. such a pain 😅

Comment (jharenza) [reply analyses/molecular-subtyping-HGG/run-molecular-subtyping-HGG_003_008.sh]: :P

**Comment by jaclyn-taroni** [comment]
Looks like everyone's comments were addressed and there are no longer any unexpected changes of consequence, so I will merge this.

**Pull request merged**